test/integration_test/applicationbackup_test.go

```diff
@@ -176,16 +181,106 @@ func createBackupLocation(
 			SecretConfig: newSecret.Name,
 		},
 	}
-	_, err = k8s.Instance().CreateBackupLocation(backupLocation)
-	return err
+	bkpLocation, err := k8s.Instance().CreateBackupLocation(backupLocation)
+	return bkpLocation, err
+}
+
+func createApplicationRestore(
+	t *testing.T,
+	name string,
+	namespace string,
+	backup *storkv1.ApplicationBackup,
+	backupLocation *storkv1.BackupLocation,
+) (*storkv1.ApplicationRestore, error) {
+	namespaceMapping := map[string]string{namespace: namespace}
+
+	appRestore := &storkv1.ApplicationRestore{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: storkv1.ApplicationRestoreSpec{
+			BackupName:       backup.Name,
+			BackupLocation:   backupLocation.Name,
+			NamespaceMapping: namespaceMapping,
+		},
+	}
+	applicationRestore, err := k8s.Instance().CreateApplicationRestore(appRestore)
+	return applicationRestore, err
+}
+
+func createApplicationBackupWithAnnotation(
+	t *testing.T,
+	name string,
+	namespace string,
+	backupLocation *storkv1.BackupLocation,
+) (*storkv1.ApplicationBackup, error) {
+
+	appBackup := &storkv1.ApplicationBackup{
+		ObjectMeta: meta.ObjectMeta{
+			Name:        name,
+			Namespace:   namespace,
+			Annotations: generateTimestampAnnotationMap(backupSyncAnnotation),
+		},
+		Spec: storkv1.ApplicationBackupSpec{
+			Namespaces:     []string{namespace},
+			BackupLocation: backupLocation.Name,
+		},
+	}
+	applicationBackup, err := k8s.Instance().CreateApplicationBackup(appBackup)
+	return applicationBackup, err
+}
+
+func addAnnotationsToBackup(
+	backup *storkv1.ApplicationBackup,
+	annotations map[string]string,
+) (*storkv1.ApplicationBackup, error) {
+	if backup.Annotations == nil {
+		backup.Annotations = map[string]string{}
+	}
+	for k, v := range annotations {
+		backup.Annotations[k] = v
+	}
+	backup, err := k8s.Instance().UpdateApplicationBackup(backup)
+	return backup, err
+}
+
+func generateTimestampAnnotationMap(annotationKey string) map[string]string {
+	t := time.Now()
+	val := t.Format("20060102-01-01-01")
+	annotationMap := map[string]string{annotationKey: val}
+	logrus.Infof("Annotations created to track backup: %v", annotationMap)
+	return annotationMap
+}
+
+func getBackupFromListWithAnnotations(backupList *storkv1.ApplicationBackupList, annotationMap map[string]string) *storkv1.ApplicationBackup {
```

Comment:
Don't need to pass in all the annotations, right? Can pass in only the key that we should be looking for.
Don't need lines 259-264 then.